qa/tasks/caasp.py

```diff
@@ -0,0 +1,137 @@
+        log.debug("Executing SSH setup")
+        self.__ssh_gen_key()
+        self.__ssh_copy_priv()
+        self.__ssh_copy_pub_to_caasp()
+
+    def __ssh_gen_key(self):
+        if os.path.isfile(self.ssh_priv):
+            os.remove(self.ssh_priv)
+        if os.path.isfile(self.ssh_pub):
+            os.remove(self.ssh_pub)
+        os.system('ssh-keygen -t rsa -b 2048 -P "" -f %s' % self.ssh_priv)
+#            subprocess.check_output(['ssh-keygen', '-t', 'rsa', '-b', '2048', '-f', self.ssh_priv])
+#            res = subprocess.check_output(["ls"])
+#            log.debug(res)
+#        except subprocess.CalledProcessError as e:
+#            log.debug('error generating key')
+#            log.debug(e.output)
+
+    def __ssh_copy_priv(self):
+        log.debug("Executing __ssh_copy_priv")
+        try:
+            subprocess.check_output(['scp', self.ssh_priv, '%s:.ssh/' %
+                                     self.mgmt_remote])
+        except subprocess.CalledProcessError as e:
+            log.debug(e.output)
+
+    def __ssh_copy_pub(self, remote):
+        log.debug("Executing __ssh_copy_pub to %s" % remote.hostname)
+        cmd = "ssh-copy-id -i %s %s" % (self.ssh_pub, remote)
+        log.debug(cmd)
+        os.system(cmd)
+ #        try:
+ #            subprocess.check_output(
+ #                ['ssh-copy-id', '-i', self.ssh_pub, '%s' % remote])
+ #        except subprocess.CalledProcessError as e:
+ #            log.debug(e.output)
+
+    def __ssh_copy_pub_to_caasp(self):
+        for i in range(sum(1 for x in all_roles_of_type(
+                self.ctx.cluster, 'caasp_master'))):
+            remote = get_remote_for_role(
+                self.ctx, "caasp_master." + str(i))
+            self.__ssh_copy_pub(remote)
+        for i in range(sum(1 for x in all_roles_of_type(
+                self.ctx.cluster, 'caasp_worker'))):
+            remote = get_remote_for_role(
+                self.ctx, "caasp_worker." + str(i))
+            self.__ssh_copy_pub(remote)
+
+    def __create_cluster(self):
+        master_remote = get_remote_for_role(self.ctx, "caasp_master.0")
+        commands = [
+            "ssh-add -L",
+            "skuba cluster init --control-plane {} cluster".format(master_remote.hostname),
+            "cd cluster && skuba node bootstrap --user ubuntu --sudo --target {} my-master".format(
+                master_remote.hostname),
+        ]
+        for command in commands:
+            self.mgmt_remote.sh("%s %s" % (self.set_agent, command))
+        for i in range(sum(1 for x in all_roles_of_type(
+                self.ctx.cluster, 'caasp_worker'))):
+            worker_remote = get_remote_for_role(
+                self.ctx, "caasp_worker." + str(i))
+            command = "cd cluster;skuba node join --role worker --user ubuntu --sudo --target {} worker.{}".format(
+                worker_remote.hostname, str(i))
+            self.mgmt_remote.sh("%s %s" % (self.set_agent, command))
```

Comment:
You do not need this agent stuff in correctly configured cluster, make sure you've setup the authorized keys correctly.